### DISCUSSION TITLE
QSocketNotifier error -- how to fix?

### DISCUSSION THREAD

**Discussion by martinellison** [slint-ui/slint#4220] [Q&A]
I have coded a simple Slint app (just a copy of the Hello World example) and I am getting an error message.

Can anyone help me with this?

The error is: `QSocketNotifier: Can only be used with threads started with QThread`.

This is on Fedora Linux using Gnome Classic with Wayland, coded in Rust.

(Presumably I am using Gnome and therefore not Qt).

*Edit: I've noticed that a window has actually appeared,,, does this mean I should ignore the error message?*

**Comment by ogoffart** [comment]
Your application is using the Qt backend.

You can disable the Qt backend in the cargo feature, or with the SLINT_BACKEND=winit runtime env variable.

It may or may not be safe to ignore the warning. I don't know what causes it.
In order to find out, you can use the `QT_FATAL_WARNINGS=1` env variable to get a crash when this happens and if you run with gdb you can get a backtrace. This would help finding out what is using QSocketNotifier and in which thread.
